libsolidity/codegen/ABIFunctions.h

```diff
@@ -66,6 +66,16 @@ class ABIFunctions
 		bool _encodeAsLibraryTypes = false
 	);
 
+	/// @returns name of an assembly function to ABI-decode values of @a _types
+	/// into memory. If @a _fromMemory is true, decodes from memory instead of
+	/// from calldata.
+	/// Can allocate memory.
+	/// Inputs: <source_offset> <source_end> (layout reversed on stack)
```

Comment:
These are the inputs if `_fromMemory` is true, otherwise no inputs?

Reply:
these are either pointers to calldata or to memory.

Reply:
Also, are those both absolute pointers or is `source_end` actually `source_len`?

Reply:
Both are absolute pointers.